src/plugincollection.js

```diff
@@ -208,6 +208,21 @@ export default class PluginCollection {
 	}
 
 	/**
+	 * Destroys all loaded plugins.
+	 *
+	 * @returns {Promise}
+	 */
+	destroy() {
+		const promises = [];
+
+		for ( const [ , pluginInstance ] of this ) {
```

Comment:
First of all: this could be simplified as: `Array.from( this._plugins.values() ).map( plugin => plugin.destroy() )`

Second, `destroy()` is optional in `PluginInterface`.